evalml/pipelines/components/transformers/preprocessing/decomposer.py

```diff
@@ -26,10 +32,122 @@ def __init__(self, parameters=None, component_obj=None, random_seed=0, **kwargs)
             **kwargs,
         )
 
+    def _set_time_index(self, X, y):
```

Comment:
Shifted `_set_time_index()` up from `PolynomialDecomposer` class to the parent since it can probably be reused.